### PR TITLE
Fixed #1266 Bug Report When a new post is added, the latest pos…

### LINKED ISSUE #1266: Bug Report When a new post is added, the latest post section in the organization's dashboard becomes distorted.
**Describe the bug**

When a new post is added, the latest section in the organization's dashboard becomes distorted.

**To Reproduce**

1. Go to any organization
2. Add some posts.
3. Go to dashboard.
4. In the lastest post section we can see the distorted design.

**Expected behavior**

**Actual behavior**
A clear and concise description of how the code performed w.r.t expectations.

**Screenshots**
Pasted image

**Additional details**

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by gauravsingh94**
…t section in the organization's dashboard becomes distorted.

**What kind of change does this PR introduce?**

**Issue Number:**

Fixes #1266 

**Did you add tests for your changes?**

**Snapshots/Videos:**

[Screencast from 2023-12-30 20-10-30.webm](https://github.com/PalisadoesFoundation/talawa-admin/assets/99260988/c24c20d7-737f-402b-ac31-363e746011cc)

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Comment by aashimawadhwa** [comment]
this is not much appealing, can we hide the name of the post creator on smaller screen sizes? @gauravsingh94

**Comment by gauravsingh94** [comment]
> this is not much appealing, can we hide the name of the post creator on smaller screen sizes? @gauravsingh94 

Yes, I can hide the name of posts creater on smaller screen. 

I will update.

**Comment by gauravsingh94** [comment]
> this is not much appealing, can we hide the name of the post creator on smaller screen sizes? @gauravsingh94

I've attempted to conceal the 'Author' section when the div is narrower than 284 pixels, but unfortunately, it doesn't seem to be functioning as intended:

```css
@media (max-width: 284px) {
  .cardItem .rightCard {
    display: none;
  }
}
```

Afterward, I aimed to center all the content within the div and also utilized `word-break`, but I noticed it affected the title differently. When a single character is particularly lengthy, the intended behavior isn't achieved:

image


Given the circumstances, it seems the pull request (PR) I've submitted might be the most suitable option for this scenario. Alternatively, an overhaul of the entire design of the card component might be necessary.

**Comment by gauravsingh94** [comment]
> > this is not much appealing, can we hide the name of the post creator on smaller screen sizes? @gauravsingh94
> 
> I've attempted to conceal the 'Author' section when the div is narrower than 284 pixels, but unfortunately, it doesn't seem to be functioning as intended:
> 
> ```css
> @media (max-width: 284px) {
>   .cardItem .rightCard {
>     display: none;
>   }
> }
> ```
> 
> Afterward, I aimed to center all the content within the div and also utilized `word-break`, but I noticed it affected the title differently. When a single character is particularly lengthy, the intended behavior isn't achieved:
> 
> image
> 
> Given the circumstances, it seems the pull request (PR) I've submitted might be the most suitable option for this scenario. Alternatively, an overhaul of the entire design of the card component might be necessary.

@aashimawadhwa @rishav-jha-mech

**Comment by rishav-jha-mech** [comment]
@gauravsingh94 

1. Use this layout
Group 22 (1)

2. To limit the number of lines using CSS. you refer to this popular question on Stack Overflow, https://stackoverflow.com/questions/3922739/limit-text-length-to-n-lines-using-css.

PS. set the limit to number of lines to one, for both the author name and post title.

**Comment by gauravsingh94** [comment]
Please Review this one i have updated it according to the given design.

image


**Screenrecoding**

[Screencast from 2024-01-03 23-19-27.webm](https://github.com/PalisadoesFoundation/talawa-admin/assets/99260988/0a70012e-1430-4400-9735-c659a63036ac)

**Comment by rishav-devraj** [comment]
@gauravsingh94 increase the column width for post section, design looks fine to me

**Comment by gauravsingh94** [comment]
image


@rishav-devraj  is it ok?

**Comment by rishav-jha-mech** [comment]
@gauravsingh94 no its not, it should take more width where it is currently

**Comment by gauravsingh94** [comment]
Now this one is ok?
@rishav-jha-mech 

image

**Comment by rishav-jha-mech** [comment]
@gauravsingh94 this should be the actual layout

Organization_ Dashboard (1)

**Comment by gauravsingh94** [comment]
@rishav-jha-mech 

Updated it according to the above layout you have given. 

Please review this one.


image

**Comment by gauravsingh94** [comment]
I have updated it according the above layout you have given.

Please review the new changes. @rishav-jha-mech 



image

File: src/components/OrganizationDashCards/CardItem.module.css [hunk 1]
```diff
@@ -22,7 +26,8 @@
   bottom: 0;
   left: 0;
   opacity: 0.12;
-  border-radius: 50%;
+  border-radius: 4px;
+  /* margin-right: 18px; */
```
Comment (rishav-jha-mech) [inline src/components/OrganizationDashCards/CardItem.module.css:30]: Don't include commented code into this PR

**Comment by rishav-jha-mech** [comment]
@gauravsingh94 write tests for the changes you have made, make the membership requests section similar in height as per design, rest UI wise all looks good

**Comment by gauravsingh94** [comment]
> @gauravsingh94 write tests for the changes you have made, make the membership requests section similar in height as per design, rest UI wise all looks good

Ok i will add the tests and and update the height.

**Comment by gauravsingh94** [comment]
@rishav-jha-mech  Can you please help me how i can cover the test  case for that line.I have tried but i am not able to.

**Comment by rishav-jha-mech** [comment]
@gauravsingh94 the issue is around upcomingEvents not being loaded in the screen, there must be something wrong with your mocks setup, please check that.

This is how one should debug the tests https://github.com/PalisadoesFoundation/talawa-admin/pull/1193#issuecomment-1873603505

File: src/components/OrganizationDashCards/CardItem.module.css [hunk 2]
```diff
@@ -33,22 +36,30 @@
...
+  -webkit-line-clamp: 1;
+  -webkit-box-orient: vertical;
 }
 
 .cardItem .time {
-  font-size: 0.9rem;
-  color: var(--bs-secondary);
+  font-size: 0.8rem;
+  font-weight: 300;
+  color: #808080;
```
Comment (rishav-jha-mech) [inline src/components/OrganizationDashCards/CardItem.module.css:57]: Don't use hardcoded values, use css variables from bootstrap

File: src/components/OrganizationDashCards/CardItem.module.css [hunk 3]
```diff
@@ -33,22 +36,30 @@
...
 }
 
 .cardItem .creator {
-  font-size: 1rem;
-  color: rgb(33, 208, 21);
+  font-size: 0.8rem;
+  color: black;
```
Comment (rishav-jha-mech) [inline src/components/OrganizationDashCards/CardItem.module.css:62]: Here also

**Comment by gauravsingh94** [comment]
I have update the hard coded values to the  css variables from bootstrap.

**Comment by gauravsingh94** [comment]
This is the section which is not getting covered in the tests.

```ts
: (
  upcomingEvents.map(
	(event: InterfaceQueryOrganizationEventListItem) => {
	  return (
		<CardItem
		  type="Event"
		  key={event._id}
		  time={event.startDate}
		  title={event.title}
		  location={event.location}
		/>
	  );
```


I have also updated the `mock`

```ts
export const EVENT_MOCK = [
  {
    request: {
      query: ORGANIZATION_EVENT_CONNECTION_LIST,
      variables: {
        organization_id: '6437904485008f171cf29924',
      },
    },
    result: {
      data: {
        eventsByOrganizationConnection: [
          {
            _id: '1',
            title: 'Sample Event',
            description: 'Sample Description',
            startDate: '2023-10-29T00:00:00.000Z',
            endDate: '2023-10-29T23:59:59.000Z',
            location: 'Sample Location',
            startTime: '08:00:00',
            endTime: '17:00:00',
            allDay: false,
            recurring: false,
            isPublic: true,
            isRegisterable: true,
          },
        ],
      },
    },
  },
];
```


I have added the test case for checking the that `upcoming event` get render or not.

```ts
test('Checking for Upcoming Event is getting rendered or not.', async () => {
    await act(async () => {
      render(
        <MockedProvider addTypename={false} link={link1}>
          <BrowserRouter>
            <Provider store={store}>
              <I18nextProvider i18n={i18nForTest}>
                <OrganizationDashboard />
              </I18nextProvider>
            </Provider>
          </BrowserRouter>
        </MockedProvider>
      );
    });

    await wait();
    expect(screen.getByText('Sample Event')).toBeInTheDocument();
  });
```


I am not able figure our why organization is not loading getting error.

```bash
Error Occured while loading the Organization
```
@rishav-jha-mech 
Please help me how i can fix this issue.

**Comment by rishav-jha-mech** [comment]
@gauravsingh94 check if to query the Organization, its id is taken from the url of that page, check if something is missing in the local storage while you are running this test. Are you using jest preview ?

**Comment by gauravsingh94** [comment]
> @gauravsingh94 check if to query the Organization, its id is taken from the url of that page, check if something is missing in the local storage while you are running this test. Are you using jest preview ?
> 

Yes I am using jest preview and there also I am getting error while getting organization

**Comment by palisadoes** [comment]
Closing due to inactivity

**Pull request closed without merging**